plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -107,94 +105,86 @@ data ChainIndexTxOut =
       -- | Value of the transaction output.
-      _ciTxOutValue           :: V1.Value,
+      _ciTxOutValue           :: Value,
       -- | Datum attached to the transaction output, either in full or as a
       -- hash reference. A transaction output protected by a Plutus script
       -- is guardateed to have an associated datum.
-      _ciTxOutScriptDatum     :: Either V1.DatumHash V1.Datum,
+      _ciTxOutScriptDatum     :: Either DatumHash Datum,
       -- | Optional reference script attached to the transaction output.
       -- The reference script is, in genereal, unrelated to the validator
       -- script althought it could also be the same.
-      _ciTxOutReferenceScript :: Maybe V1.Script,
+      _ciTxOutReferenceScript :: Maybe Script,
       -- | Validator protecting the transaction output, either in full or
       -- as a hash reference.
-      _ciTxOutValidator       :: Either V1.ValidatorHash V1.Validator
+      _ciTxOutValidator       :: Either ValidatorHash Validator
     }
   deriving (Show, Eq, Serialise, Generic, ToJSON, FromJSON, OpenApi.ToSchema)
 
 makeLenses ''ChainIndexTxOut
 makePrisms ''ChainIndexTxOut
 
--- | Converts a transaction output from the chain index to the plutus-ledger-api
--- transaction output.
---
--- Note that 'ChainIndexTxOut' supports features such inline datums and
--- reference scripts which are not supported by V1 TxOut. Converting from
--- 'ChainIndexTxOut' to 'TxOut' and back is therefore lossy.
-toTxOut :: ChainIndexTxOut -> V1.Tx.TxOut
-toTxOut (PublicKeyChainIndexTxOut addr v _outputDatum _referenceScript) =
-  V1.Tx.TxOut addr v Nothing
-toTxOut (ScriptChainIndexTxOut addr v (Left dh) _referenceScript _validator)     =
-  V1.Tx.TxOut addr v (Just dh)
-toTxOut (ScriptChainIndexTxOut addr v (Right d) _referenceScript _validator)     =
-  V1.Tx.TxOut addr v (Just $ datumHash d)
-
--- | Converts a plutus-ledger-api transaction output to the chain index
--- transaction output.
-fromTxOut :: V1.Tx.TxOut -> Maybe ChainIndexTxOut
-fromTxOut V1.Tx.TxOut { txOutAddress, txOutValue, txOutDatumHash } =
-  case V1.addressCredential txOutAddress of
-    V1.PubKeyCredential _ ->
-      -- V1 transactions don't support inline datums
-      pure $ PublicKeyChainIndexTxOut txOutAddress txOutValue V2.NoOutputDatum Nothing
-    V1.ScriptCredential vh ->
-      txOutDatumHash >>= \dh ->
-        pure $ ScriptChainIndexTxOut txOutAddress txOutValue (Left dh) Nothing (Left vh)
+toTxOut :: ChainIndexTxOut -> TxOut
+toTxOut (PublicKeyChainIndexTxOut addr v outputDatum referenceScript) =
+  TxOut addr v outputDatum (fmap scriptHash referenceScript)
+toTxOut (ScriptChainIndexTxOut addr v (Left dh) referenceScript _validator) =
+  TxOut addr v (OutputDatumHash dh) (fmap scriptHash referenceScript)
+toTxOut (ScriptChainIndexTxOut addr v (Right d) referenceScript _validator) =
+  TxOut addr v (OutputDatum d) (fmap scriptHash referenceScript)
+
+-- fromTxOut :: TxOut -> Maybe ChainIndexTxOut
```

Comment:
unfortunately we cannot make a `ChainIndexTxOut` out of a `TxOut` because the reference script is gone. We could return `Nothing` here but I don't thing the meaning of this function is clear at all. We need to revisit.